src/stage/stage.js

```diff
@@ -763,8 +838,12 @@ export default class Stage extends BaseStage {
             }
         };
 
+        var modee = this.mode;
```

Comment:
Just for future reference, prefer `const` or `let` over `var` (since `var` does hoisting which can do unexpected things).